app/controllers/course/video/topics_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+class Course::Video::TopicsController < Course::Video::Controller
+  load_resource :topic, class: Course::Video::Topic.name
+  before_action :load_existing_topic, except: :show
+
+  include Course::Discussion::PostsConcern
+  skip_load_and_authorize_resource :post, only: :show
+
+  def index
+    @topics = @video.topics.includes(posts: :children).order(:timestamp)
+    @topics = @topics.reject { |topic| topic.posts.empty? }
+    @posts = @topics.map(&:posts).inject(Course::Discussion::Post.none, :+)
+  end
+
+  def create
+    result = @topic.class.transaction do
+      @post.title = @video.title
+      raise ActiveRecord::Rollback unless @post.save && create_topic_subscription && update_topic_pending_status
+      raise ActiveRecord::Rollback unless @topic.save
+      true
+    end
+
+    head :bad_requested unless result
+  end
+
+  def show; end
+
+  private
+
+  def topic_params
+    params.permit(:video_id, :timestamp)
+  end
+
+  def timestamp_param
+    params[:timestamp]
+  end
+
+  def topic_param
+    params[:topic_id]
+  end
+
+  def parent_id_param
+    params[:discussion_post].present? ? params[:discussion_post][:parent_id] : nil
+  end
+
+  def discussion_topic
+    @topic.discussion_topic
+  end
+
+  def create_topic_subscription
+    @topic.ensure_subscribed_by(current_user)
```

Comment:
@allenwq when a topic is subscribed by user A, what does that mean? Is it that if user B replies to this topic, then user A gets an email that someone replies? 

If that is the case maybe we do not need subscription for video posts, as it might potentially get noisy?

Reply:
Yes, all subscribers of the topic will receive emails when there's a new reply.

Reply:
Hmm, from what I for the other discussion models, the post controller calls a `notify(post)` on the `actable` model to trigger the emails.

For `Courser::Video::Topic` this method is currently left unimplemented. Moving forward, I think we should have emails only if there is a reply to one of posts you're involved in (you made the parent post or have replied to that post).

Reply:
👍 I also took a look at the existing code, the implementation of subscription is left up to the actable. Agree with the recommendation.